src/graphql/resolvers/__tests__/user.ts

```diff
@@ -0,0 +1,254 @@
+
+//#region Initializations
+
+const dataStorage = new StorageDataSource(mock.db);
+
+//#endregion Initializations
+
+describe('Unit', () => {
+
+  describe('[Resolvers - Mutations]', () => {
+    
+
+    describe('user', () => {
+      
+
+      it('read one by Id - success', async () => {
+        mock.db.scan.mockReturnValueOnce({
+          promise: () => Promise.resolve(mockedReadByIdDbResponse)
+        });
+        const getNewId = jest.fn().mockImplementation(() => mockedDocumentId);
+        dataStorage.getNewId = getNewId.bind(dataStorage); 
+
+        await expect(userQueries.users(null, mockedReadByIdInputArgs, { dataSources: { storage: dataStorage}}, null)).resolves.toEqual(mockedReadByResponse);
+      });
+
+      it('read one by Id - failure', async () => {
+        mock.db.scan.mockReturnValueOnce({
+          promise: () => Promise.reject(new Error("read failed"))
+        });
+
+        await expect(userQueries.users(null, mockedReadByIdInputArgs, { dataSources: { storage: dataStorage}}, null)).rejects.toEqual(new ApolloError("read failed"));
+      });
+
+      it('read limited list - success', async () => {
+        mock.db.scan.mockReturnValueOnce({
+          promise: () => Promise.resolve(mockedReadLimitedListDbResponse)
+        });
+
+        await expect(userQueries.users(null, mockedReadLimitedListInputArgs, { dataSources: { storage: dataStorage}}, null)).resolves.toEqual(mockedReadLimitedListResponse);
+      });
+
+      it('read limited list - failure', async () => {
+        mock.db.scan.mockReturnValueOnce({
+          promise: () => Promise.reject(new Error("read failed"))
+        });
+
+        await expect(userQueries.users(null, mockedReadLimitedListInputArgs, { dataSources: { storage: dataStorage}}, null)).rejects.toEqual(new ApolloError("read failed"));
+      });
+    });
+    
+    describe('create service', () => {
+      it('success', async () => {
+        mock.db.put.mockReturnValueOnce({
+          promise: () => Promise.resolve()
+        });
+
+        await expect(userMutations.createUser(null, mockedCreateInputArgs, { dataSources: { storage: dataStorage}}, null)).resolves.toEqual(mockedCreateResponse);
+      });
+
+      it('failure', async () => {
+        mock.db.put.mockReturnValueOnce({
+          promise: () => Promise.reject(new Error("creation failed"))
+        });
+
+        await expect(userMutations.createUser(null, mockedCreateInputArgs, { dataSources: { storage: dataStorage}}, null)).rejects.toEqual(new ApolloError("creation failed"));
+      });
+    });
+
+    describe('update service', () => {
+      it('success', async () => {
+        mock.db.update.mockReturnValueOnce({
+          promise: () => Promise.resolve(mockedUpdateResponse)
+        });
+
+        await expect(userMutations.updateUser(null, mockedUpdateInputArgs, { dataSources: { storage: dataStorage}}, null)).resolves.toEqual(mockedUpdateResponse.Attributes);
+      });
+
+      it('failure', async () => {
```

Comment:
What is the reason this test should fail?

Reply:
Because the StorgeDataSources.ts 'update' function catch the exception the dymamoDb client throw in those cases in which an expresion for update is executed for and non existing id. 
The catch block throws an ApilloError with '"Document not found" message. This is how I designed it.
So, based on that, the revolver for this escenarion will reject the promise with such error.